kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -107,6 +107,10 @@ def before_command_run(
             if not project_metadata:  # in package mode
                 return
 
+            for arg in command_args:
```

Comment:
Doesn't this mean though that using the `--telemetry` flag is the same as opting out? If we return at this point, no data will be sent even if the user provided "yes" to the flag.

Reply:
Hmm I imagined the person  would only use the flag once and then on subsequent runs that would be handled, but you're right. I'm going to look over this one again.